runtime/lua/vim/filetype.lua

```diff
@@ -29,8 +29,69 @@ local function getline(bufnr, start_lnum, end_lnum)
   return table.concat(lines) or ''
 end
 
--- Filetypes based on file extension
+--- Get a single line or line-range from the buffer.
+---
+---@param bufnr number|nil The buffer to get the lines from
+---@param start_lnum number The line number of the first line (inclusive, 1-based)
+---@param end_lnum number|nil The line number of the last line (inclusive, 1-based)
+---@return table<string>|string Array of lines, or string when end_lnum is omitted
+function M.getlines(bufnr, start_lnum, end_lnum)
```

Comment:
I think all of these should be marked `—--@private` so that they are not included in the docs. They should only be used for filetype detection, in which case people can see the doc comments in the source code.